tests/test_mc_simulator.py

```diff
@@ -610,3 +612,228 @@ def test_MantarrayMcSimulator__allows_status_bits_to_be_set_through_testing_queu
+
+
+def test_MantarrayMcSimulator__discards_commands_from_pc_during_reboot_period__and_sends_reboot_response_packet_before_reboot__and_sends_status_beacon_after_reboot(
+    mantarray_mc_simulator, mocker
+):
+    simulator = mantarray_mc_simulator["simulator"]
+
+    spied_get_cms_since_init = mocker.spy(simulator, "get_cms_since_init")
+    spied_randint = mocker.spy(random, "randint")
+
+    reboot_duration = 5
+    reboot_times = [4, reboot_duration]
+    mocker.patch.object(
+        mc_simulator,
+        "_get_secs_since_reboot_command",
+        autospec=True,
+        side_effect=reboot_times,
+    )
+
+    spied_reset = mocker.spy(simulator, "_reset_start_time")
+
+    # remove initial status beacon
+    invoke_process_run_and_check_errors(simulator)
+    initial_status_beacon_length = STATUS_BEACON_SIZE_BYTES - spied_randint.spy_return
+    initial_status_beacon = simulator.read(size=initial_status_beacon_length)
+    assert len(initial_status_beacon) == initial_status_beacon_length
+
+    # send reboot command
+    dummy_timestamp = 0
+    test_reboot_command = create_data_packet(
+        dummy_timestamp,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_SIMPLE_COMMAND_PACKET_TYPE,
+        bytes([SERIAL_COMM_REBOOT_COMMAND_BYTE]),
+    )
+    simulator.write(test_reboot_command)
+    invoke_process_run_and_check_errors(simulator)
+
+    # test that reboot response packet is sent
+    expected_reboot_response = create_data_packet(
+        spied_get_cms_since_init.spy_return,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_SIMPLE_COMMAND_PACKET_TYPE,
+        bytes(0),
+    )
+    reboot_response = simulator.read(size=len(expected_reboot_response))
+    assert reboot_response == expected_reboot_response
+
+    # test that handshake is ignored
+    test_handshake = create_data_packet(
+        dummy_timestamp,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_HANDSHAKE_PACKET_TYPE,
+        bytes(0),
+    )
+    simulator.write(test_handshake)
+    invoke_process_run_and_check_errors(simulator)
+    response_during_reboot = simulator.read(size=HANDSHAKE_RESPONSE_SIZE_BYTES)
+    assert len(response_during_reboot) == 0
+
+    # test that status beacon is sent after reboot
+    invoke_process_run_and_check_errors(simulator)
+    expected_beacon = create_data_packet(
+        spied_get_cms_since_init.spy_return,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_STATUS_BEACON_PACKET_TYPE,
+        DEFAULT_SIMULATOR_STATUS_CODE,
+    )
+    status_beacon = simulator.read(size=len(expected_beacon))
+    assert status_beacon == expected_beacon
+
+    # test that start time was reset
+    spied_reset.assert_called_once()
+
+
+def test_MantarrayMcSimulator__reset_status_code_after_rebooting(
+    mantarray_mc_simulator_no_beacon, mocker
+):
+    simulator = mantarray_mc_simulator_no_beacon["simulator"]
+    testing_queue = mantarray_mc_simulator_no_beacon["testing_queue"]
+
+    reboot_duration = 5
```

Comment:
should this `5` be tied to something in `constants.py` that's used in both the tests and the main code?  it looks like we have two sources of "truth" here instead of having a single source of truth